### ISSUE TITLE
Add xtest(s) to detect method signatures that are inconsistant with source/Roast

### ISSUE THREAD

**Issue by codesections** [Raku/doc#3604]
## Problem

I have noticed a number of places where there are slight typos/errors in the way we document method signatures.  For example, there are several methods that are missing a `:` after the invocant (e.g., [`Str.unival`](https://docs.raku.org/routine/unival#(Str)_method_unival) (which I haven't yet fixed so that I can use it as an example here – I'll fix it soon, though)

## Suggestions

We should have one or more tests that help detect these errors.  Some of these may be traditional xtests that, in general, should not fail unless there's a problem in the docs.

Others, however, might be more along the lines of test diagnostics that could identify _potential_ errors, but that might be a bit over-inclusive and thus would be expected to have at least a few failures (e.g., we could detect all methods that don't declare an invocant as their first parameter.  The _vast_ majority of methods do declare an invocant, so it would be helpful to know about the ones that that don't.  Some of those might be errors, but some might be the rare case where we intentionally don't declare an invocant.  These tests could also be useful, but might not belong in `doc/xt`.

**Comment by JJ** [comment]
There's missing-methods.t and list-missing-methods.t These will probably
catch those errors, right? Although we might want to check that one,
because `unival` is not caught when run with... raku
util/list-missing-methods.p6 doc/Type/Str.pod6
Also, syntax errors should be caught with examples-compilation. Is that
what you mean?

**Comment by codesections** [comment]
> There's missing-methods.t and list-missing-methods.t These will probably
catch those errors, right? … Also, syntax errors should be caught with examples-compilation. Is that
what you mean?

The scope I have in mind for these tests is slightly different from the scope for `list-missing-methods` and `examples-compilation`.  Consider the `Str.unival` method documentation again.  The docs define it as

```raku
    multi method unival(Str:D --> Numeric)
```

This is a definition for the `unival` method, so `list-missing-method` will (correctly!) not list the `unival` method as missing.  Similarly, it's a _syntactically valid_ method definition (specifically, it uses an implicit invocant and an unnamed positional parameter; that is, it's equivalent to `multi method unival(Str: Str:D $, --> Numeric)`) so `examples-compilation` will (correctly!) report that it compiles ok.  It's just not the method signature we _intended_ to document/doesn't match reality.

I plan to have a WIP PR that implements a basic test soon, which should help clarify the types of issues that similar tests might catch.

**Comment by JJ** [comment]
If list-missing-methods does not do that, maybe it should... Or maybe you'd
want to wrap these old utils (which are utils, not xt) into the new ones...

**Comment by codesections** [comment]
> these old utils

Well, `list-missing-methods` isn't **that** old, given that it's the script I rewrote just a couple of weeks ago :grin: 

We certainly could wrap these sorts of test utilities up into something larger (and I still am planning to spin `list-missing-methods` off into its own module, as we discussed).  However, I think there's value in keeping a conceptual distinction for looking for _missing_ documentation versus checking the accuracy of documentation.  Both are really important, but combining them could risk makin both the test code and the output of the tests more confusing.
